aten/src/ATen/cpu/vec/vec512/vec512_double.h

```diff
@@ -0,0 +1,454 @@
+#pragma once
+
+// DO NOT DEFINE STATIC DATA IN THIS HEADER!
+// See Note [Do not compile initializers with AVX]
+
+#include <ATen/cpu/vec/vec512/intrinsics.h>
+#include <ATen/cpu/vec/vec512/vec512_base.h>
+#if (defined(CPU_CAPABILITY_AVX512)) && !defined(_MSC_VER)
+#include <sleef.h>
+#endif
+
+namespace at {
+namespace vec {
+// See Note [Acceptable use of anonymous namespace in header]
+namespace {
+
+#if defined(CPU_CAPABILITY_AVX512) && !defined(_MSC_VER)
+
+template <> class Vectorized<double> {
+private:
+  static constexpr __m512i zero_vector {0, 0, 0, 0, 0, 0, 0, 0};
+public:
+  // values needs to be public for compilation with clang
+  // as vec512.h uses it

```

Comment:
weird

Reply:
Yes, that's strange. This comment hasn't been copied from `vec256_double.h`.
I'll check what's going on here.

Reply:
I don't see this comment in the existing `vec256` code, so I'll check if (and why) compilation actually fails with `clang`, if `values` is private. I should've mentioned the complete context in the comment. Lesson learnt.